### ISSUE TITLE
double-take with unifi nvr and 

### ISSUE THREAD

**Issue by kingunlim** [jakowenko/double-take#229]
Hello, 
i wanted to ask if there is maybe a possibility to use double-take with unifi protect?
i saw that someone managed to make a addon for mqtt and also recently updated it to make use of the smart detections:
https://github.com/terafin/mqtt-unifi-protect-bridge

<img width="1013" alt="image" src="https://user-images.githubusercontent.com/42124176/179716216-266b8be6-c3bc-4a7b-ab0b-3bdab00726e3.png">

i saw that double take forces to use frigate, so i would think it would some tweaks from your site?

or maybe you have a other idea to realize this?

best regards

**Comment by jakowenko** [comment]
If you want to go the MQTT route, you'll need to find an integration that publishes a jpeg encoded image. This is what Frigate does and how Double Take processes the MQTT image.

The [cameras](https://github.com/jakowenko/double-take#cameras) part of the docs show how you can setup a camera to listen for a custom topic. The mqtt-unifi-protect-bridge topics seem more related to the state of the sensor rather than the actual image from the camera.

**Comment by jakowenko** [comment]
@kingunlim, the docs here https://github.com/hjdhjd/homebridge-unifi-protect/blob/main/docs/MQTT.md#homekit-support-for-the-unifi-protect-ecosystem-using-homebridge show that it's possible to publish snapshot events, including publishing the actual images over MQTT. So if you can get that working, then Double Take should be able to process those.

**Comment by kingunlim** [comment]
thanks for your answers!

Hmm... so the easiest route would be to try homebridge, i can configure it and then i will see how that works, atm i use the scrypted docker to bring my unifi cameras to homekit secure video, but if double take works together with homebridge for the snapshot feature than i will change to homebridge.
do you also see the other addons i post, just wanted to ask if you see something with a snapshot feature there?

thanks and best regards

**Comment by kingunlim** [comment]
@jakowenko 
i configured the homebridge addon + mqtt broker, 
maybe you can give me a advice how the config file for double-take would need to look like to work and listen to the mqtt events?
as i said, i would think that i don't need frigate for this? because the smart alerts should be carried over to mqtt via the homebridge addon?
<img width="1348" alt="image" src="https://user-images.githubusercontent.com/42124176/181013571-58aec9e6-e6a7-496f-9861-b4f87c76fb18.png">

**Comment by NickM-27** [comment]
Would need to see the MQTT topics and then can suggest from there

**Comment by kingunlim** [comment]
alright, so the log from the mosquitto broker from homeassistant would be enough?
I'll post them when i come home from work ok?

**Comment by NickM-27** [comment]
I think so, that or MQTT explorer view

**Comment by kingunlim** [comment]
alright, checked my config again, and i have some troubles for the connection from my homebridge (extra raspberry) to my homeassistant (mqtt broker)- also extra raspberry.
now everything works, but i think i need to configure the snapshot function somewhere?
is it right to conifgure it here?
<img width="592" alt="image" src="https://user-images.githubusercontent.com/42124176/181049865-0e7610e2-c46f-491a-a232-24ecf30cbabd.png">
so, for instance i asume: 
<img width="594" alt="image" src="https://user-images.githubusercontent.com/42124176/181050105-a4f59677-89b1-45cf-ba6a-392c41fdd5b6.png">
but as i write i would think thats not the right place, so i assume somewhere in the config from the homebridge unfi protect plugin?

also i don't know if its neccessary to configure for example one cam via homekit controller - like my appletv or homeassistant, so that it works at all?

**Comment by kingunlim** [comment]
and maybe someone can guide me along, how to prober config it after mqtt, 
so i think the developer says i can use the local snapshot that my unifi protect smart system triggers and homebridge will then listen for this event and triggers the snapshot - now double-take must somewhere listen to this special event and then create the picture, so i need a detector?

**Comment by kingunlim** [comment]
maybe someone is willing to help me out here?

**Comment by townsmcp** [comment]
Just saw this thread and the Unifi HomeBridge update with HKSV but more importantly the HomeBridge plugin works with MQTT including saving the motion image to MQTT ([see here](https://github.com/hjdhjd/homebridge-unifi-protect/blob/main/docs/MQTT.md)). I am about to order some Unifi kit (doorbell and G4 Instant to replace Arlo HD Pro 2 and Nest Hello) so I’ve been trying to do research into see if having a Unifi doorbell will be quicker person detection and recognition rather than the Nest Hello into Frigate and then out to another tool and utilising Double Take to glue everything together. I’ve also noticed DeepStack has gone

**Comment by kingunlim** [comment]
> Just saw this thread and the Unifi HomeBridge update with HKSV but more importantly the HomeBridge plugin works with MQTT including saving the motion image to MQTT ([see here](https://github.com/hjdhjd/homebridge-unifi-protect/blob/main/docs/MQTT.md)). I am about to order some Unifi kit (doorbell and G4 Instant to replace Arlo HD Pro 2 and Nest Hello) so I’ve been trying to do research into see if having a Unifi doorbell will be quicker person detection and recognition rather than the Nest Hello into Frigate and then out to another tool and utilising Double Take to glue everything together. I’ve also noticed DeepStack has gone

hey did you find a way to implement it?

**Comment by townsmcp** [comment]
@kingunlim no. I installed the HomeBridge plugin however it’s performance was not great. It is a few seconds behind live performance. I have now added the G4 Instant to Scrypted which is instant performance. There is an MQTT plugin for Scrypted which I want to have a play with - I’m not 100% sure yet if it saves the jpeg as a blob or just the event. 
However as it stands right now, I have Scrypted rebroadcasting to Frigate which is amazing response (real-time). As I have a Coral M.2 installed in the laptop with Proxmox, inference dropped from 120ms to around 6ms. Today I installed Compreface and Double Take. It is all working very well together so far
